tests/testthat/test-translate_sql.R

```diff
@@ -133,3 +133,173 @@ with_locale(test.locale(), test_that)('as.<type>() works', {
     dplyr::sql('CAST("x" AS VARBINARY)')
   )
 })
+
+with_locale(test.locale(), test_that)('`[[` works for char/numeric indices', {
+  dbplyr_version <- try(utils::packageVersion('dbplyr'))
+  if (inherits(dbplyr_version, 'try-error')) {
+    skip('dbplyr not available')
+  } else if (dbplyr_version < '1.4.0') {
+    skip('remote evaluation of `[[` requires dbplyr >= 1.4.0')
+  }
+
+  translate_sql <- RPresto:::dbplyr_compatible('translate_sql')
+  s <- setup_mock_dplyr_connection()[['db']]
+
+  # a character index should be escaped
+  expect_equal(
+    translate_sql(x[['a']], con=s[['con']]),
+    dbplyr::build_sql(dbplyr::ident('x'), "['a']", con=s[['con']])
```

Comment:
Let's use the character equivalents in all of these instead of building them with `build_sql`, e.g. `"x"['a']`.

Reply:
Done.